### ISSUE TITLE
[Bug]XAML binding behavoir changed V4.1->V4.2

### ISSUE THREAD

**Issue by sude22** [xamarin/Xamarin.Forms#7376]
### Description
See attached project to reproduce the problem:
With Xamarin.Forms V4.2.0.709249, the XAML binding behavoir changed when using 
`<StackLayout x:Name="stacklayout" Orientation="Vertical" IsEnabled="{Binding IsDataInputEnabled}">`
and
`<x1:ColorSelectorControl BindingContext="{Binding ColorSelector1}" IsEnabled="{Binding Source={x:Reference stacklayout},Path=IsEnabled}" />`

When "IsDataInputEnabled" is initial set to false, starting with V4.2 the button inside the color picker control is not shown (gray) as disabled. This works with V4.1

Please try with the attached project.
[BindingProblem01.zip](https://github.com/xamarin/Xamarin.Forms/files/3574130/BindingProblem01.zip)

### Steps to Reproduce

1. Build project with reference to Xamarin.Forms V4.2.0.709249
2. Start UWP app, the color picker control is disabled, but the button inside is not shown as disabled (grey)
3. Toggle the switch and the control and the button inside is correctly dis/enabled
4. Build project with reference to Xamarin.Forms 4.1.0.709244
5. Start UWP app, the color picker control is disabled, and the button inside is shown as disabled (grey), that's OK.

### Reproduction Link

[BindingProblem01.zip](https://github.com/xamarin/Xamarin.Forms/files/3574132/BindingProblem01.zip)

**Comment by StephaneDelcroix** [comment]
most likely a dupe of #7286 or #7097, should be fixed by #7356. regression caused by #7040

verify that this is actually fixed by #7356 before closing.

**Comment by samhouts** [comment]
@sude22 I'm having trouble getting this to build. It doesn't recognize the MVVM Light library. Can you please confirm that it's set up properly? Thanks!

**Comment by sude22** [comment]
@samhouts you could try to remove the mvvmlight nuget package and and add it again, that should work.

**Comment by PureWeen** [comment]
@sude22 I think you're supposed to use MvvmLightLibsStd10 for netstandard libraries

Also the netstandard CSPROJ file has a hard reference to a local dll which might be why you are seeing different behavior then we are when running this project

**Comment by sude22** [comment]
@PureWeen you are right, I removed the hard reference and added MVVMLightLibsStd10, now it should work, sorry.

[BindingProblem01.zip](https://github.com/xamarin/Xamarin.Forms/files/3586354/BindingProblem01.zip)

**Comment by sude22** [comment]
Just updated the BindingProblem01 project to the 4.2.0 Service Release 1 (4.2.0.778463), but the problem still occures. 
I thought it might be fixed with #7356 because this bug was referenced in #7286
